ios/RNPurchases.m

```diff
@@ -328,7 +328,7 @@ static void logUnavailablePresentCodeRedemptionSheet() {
 
 RCT_EXPORT_METHOD(beginRefundRequestForActiveEntitlement:(RCTPromiseResolveBlock)resolve
                                                   reject:(RCTPromiseRejectBlock)reject) {
-    #if TARGET_OS_IPHONE
+    #if defined(TARGET_OS_IPHONE) && !defined(TARGET_OS_MACCATALYST)
```

Comment:
If `defined(TARGET_OS_IPHONE)` is required, at least we also need to check `#if TARGET_OS_IPHONE`, since `TARGET_OS_IPHONE` might be `0`.